lib/sync/fetcher.go

```diff
@@ -159,10 +161,21 @@ func (f *BlockFetcher) fetch(ctx context.Context, si *SyncInfo) error {
 		if err := json.Unmarshal(bt.Message, &tx); err != nil {
 			return err
 		}
+		txmap[bt.Hash] = &tx
+	}
+
+	for _, hash := range blk.Transactions {
+		tx, ok := txmap[hash]
+		if !ok {
+			//TODO(anarcher): Error type for controlling timeout
+			err := fmt.Errorf("Tx: %s not found in block height %d", hash, height)
+			return err
+		}
+		si.Txs = append(si.Txs, tx)
 
-		si.Txs = append(si.Txs, &tx)
 	}
 
+	//TODO(anarcher): Remove this comments
```

Comment:
How about just removing this and below comments?

Reply:
Ok! I will do it :->